assets/js/googlesitekit/data/utils.test.js

```diff
@@ -124,6 +128,231 @@ describe( 'data utils', () => {
+				resolvers: collectResolvers( storeOne.resolvers, storeTwo.resolvers ),
+				selectors: collectSelectors( storeOne.selectors, storeTwo.selectors ),
+			};
+			expect( JSON.stringify( combineStores( storeOne, storeTwo ) ) ).toBe( JSON.stringify( expectedCombinedStore ) );
+		} );
+
+		it( 'should modify combined state', () => {
+			const combinedStore = combineStores( storeOne, storeTwo );
+			let state = combinedStore.reducer();
+
+			expect( state ).toEqual( { one: 1, two: 2 } );
+
+			// It should still respond to the original actions.
+			state = combinedStore.reducer( state, combinedStore.actions.fakeActionOne() );
+			expect( state ).toEqual( { one: true, two: 2 } );
+
+			state = combinedStore.reducer( state, combinedStore.actions.fakeActionTwo() );
+			expect( state ).toEqual( { one: true, two: 2 } );
+		} );
+
+		it( 'should combine stores missing any of the keys', () => {
+			const combinedStoreMissing = {
+				INITIAL_STATE: collect( storeOne.INITIAL_STATE, storeTwo.INITIAL_STATE, storeThree.INITIAL_STATE ),
+				actions: collectActions( storeOne.actions, storeTwo.actions ),
+				controls: collectControls( storeOne.controls, storeTwo.controls ),
+				reducer: collectReducers( storeOne.reducer, storeTwo.reducer ),
+				resolvers: collectResolvers( storeOne.resolvers, storeTwo.resolvers ),
+				selectors: collectSelectors( storeOne.selectors, storeTwo.selectors ),
+			};
+			expect( JSON.stringify( combineStores( storeOne, storeTwo, storeThree ) ) ).toBe( JSON.stringify( combinedStoreMissing ) );
+
+			const combinedStoreMissingTwo = {
+				INITIAL_STATE: collect( storeOne.INITIAL_STATE, storeTwo.INITIAL_STATE ),
+				actions: collectActions( storeOne.actions, storeTwo.actions ),
+				controls: collectControls( storeOne.controls, storeTwo.controls ),
+				reducer: collectReducers( storeOne.reducer, storeTwo.reducer, storeFour.reducer ),
+				resolvers: collectResolvers( storeOne.resolvers, storeTwo.resolvers ),
+				selectors: collectSelectors( storeOne.selectors, storeTwo.selectors ),
+			};
+			expect( JSON.stringify( combineStores( storeOne, storeTwo, storeFour ) ) ).toBe( JSON.stringify( combinedStoreMissingTwo ) );
+		} );
+
+		it( 'INITIAL_STATEs, reducers, and actions should all work together even if provided by separate incomplete stores', () => {
+			const combinedStore = combineStores(
+				{ INITIAL_STATE: { one: 1 } },
+				{ reducer: combineStoresFakeReducerOne },
+				storeTwo,
+				storeFive
+			);
+			let state = combinedStore.reducer();
+
+			expect( state ).toEqual( { one: 1, two: 2 } );
+
+			// Reducer from one incomplete store responds to an action added by a different incomplete store.
+			state = combinedStore.reducer( state, combinedStore.actions.fakeActionOne() );
+			expect( state ).toEqual( { one: true, two: 2 } );
+		} );
+
+		it( 'should error if action keys are duplicated', () => {
+			expect( () => {
+				combineStores( storeOne, storeFive );
```

Comment:
As @aaemnnosttv mentioned: without these stores being defined inside this test it's difficult to reason about what's happening. I'm _assuming_ that `storeOne` and `storeFive` have duplicated action names, but I can't tell without tracking them down elsewhere in the file. I think it'd be good to put those definitions inline.

A good aim for tests is to optimise for readability and to ignore the desire to "not repeat yourself". Tests—especially their fixtures/variables—should _frequently_ be repeated if needed. We'll only be writing a unit test once—and with a public, frequently-used API like this it's unlikely to change frequently. We should make it as easy-as-possible to reason about a single broken test 😄

Reply:
Yep, that makes sense. I'll go through and make these more collocated and readable. Thanks!